tools/clang/unittests/HLSL/PixDiaTest.cpp

```diff
@@ -2755,4 +2756,159 @@ void main()
+};
+
+RWStructuredBuffer<float4> floatRWUAV: register(u0);
+
+namespace StressScopesABit
+{
+#include "included.h"
+}
+
+namespace StressScopesMore
+{
+float4 InlinedFunction(in BuiltInTriangleIntersectionAttributes attr, int offset)
+{
+  float4 loadedColor = floatRWUAV.Load(offset + attr.barycentrics.x + 42);
+  float4 color2 = StressScopesABit::StressScopesEvenMore::DeeperInlinedFunction(attr, offset) + loadedColor;
+  float4 color3 = StressScopesABit::StressScopesEvenMore::DeeperInlinedFunction(attr, offset+1);
+  return color2 + color3;
+}
+}
+
+[shader("closesthit")]
+void ClosestHitShader0(inout RayPayload payload, in BuiltInTriangleIntersectionAttributes attr)
+{
+    payload.color = StressScopesMore::InlinedFunction(attr, 0);
+}
+
+
+[shader("closesthit")]
+void ClosestHitShader1(inout RayPayload payload, in BuiltInTriangleIntersectionAttributes attr)
+{
+    payload.color = StressScopesMore::InlinedFunction(attr, 1);
+}
+)";
+
+  CComPtr<DxcIncludeHandlerForInjectedSourcesForPix> pIncludeHandler =
+      new DxcIncludeHandlerForInjectedSourcesForPix(this, {{L"included.h",
+                                                            R"(
+
+namespace StressScopesEvenMore
+{
+float4 DeeperInlinedFunction(in BuiltInTriangleIntersectionAttributes attr, int offset)
+{
+  float4 ret = float4(0,0,0,0);
+  for(int i =0; i < offset; ++i)
+  {
+    float4 color0 = floatRWUAV.Load(offset + attr.barycentrics.x);
+    float4 color1 = floatRWUAV.Load(offset + attr.barycentrics.y);
+    ret += color0 + color1;
+  }
+  return ret;
+}
+}
+)"}});
+
+  auto dxilDebugger =
+      CompileAndCreateDxcDebug(hlsl, L"lib_6_6", pIncludeHandler);
+
+  struct SourceLocations {
+    CComBSTR Filename;
+    DWORD Column;
+    DWORD Line;
+  };
+
+  std::vector<SourceLocations> sourceLocations;
+  DWORD instructionOffset =
+      AdvanceUntilFunctionEntered(dxilDebugger, 0, L"ClosestHitShader0");
+  instructionOffset = AdvanceUntilFunctionEntered(
+      dxilDebugger, instructionOffset, L"DeeperInlinedFunction");
+  DWORD RegisterNumber0 = GetRegisterNumberForVariable(
+      dxilDebugger, instructionOffset, L"ret", L"x");
+  instructionOffset = AdvanceUntilFunctionEntered(
+      dxilDebugger, instructionOffset, L"InlinedFunction");
+  DWORD RegisterNumber2 = GetRegisterNumberForVariable(
+      dxilDebugger, instructionOffset, L"color2", L"x");
+  instructionOffset = AdvanceUntilFunctionEntered(
+      dxilDebugger, instructionOffset, L"ClosestHitShader1");
+  instructionOffset = AdvanceUntilFunctionEntered(
+      dxilDebugger, instructionOffset, L"DeeperInlinedFunction");
+  DWORD RegisterNumber1 = GetRegisterNumberForVariable(
+      dxilDebugger, instructionOffset, L"ret", L"x");
+  instructionOffset = AdvanceUntilFunctionEntered(
+      dxilDebugger, instructionOffset, L"InlinedFunction");
+  DWORD RegisterNumber3 = GetRegisterNumberForVariable(
+      dxilDebugger, instructionOffset, L"color2", L"x");
+  VERIFY_ARE_NOT_EQUAL(RegisterNumber0, RegisterNumber1);
```

Comment:
Can we also test that the register numbers are not -1?

Reply:
Sure. Any -1 will eventually cause the final test to fail, but it makes sense to test early and often: more debuggable and readable, at least.

Reply:
If one of `RegisterNumber0` or `RegisterNumber1` is -1, the test will just pass.